### DISCUSSION TITLE
getting current state to manipulate

### DISCUSSION THREAD

**Discussion by EdgarsLv** [reduxjs/redux-toolkit#2927] [Q&A]
I'm converting my react js app to ts.

In js this works fine. How to do that in ts?
![image](https://user-images.githubusercontent.com/56837461/202910793-47e993b7-df73-46c5-bb0e-1e266add5d72.png)

If i change filter to map, then it works. Why it doesn't works with filter?
![image](https://user-images.githubusercontent.com/56837461/202911286-c70a759f-fc92-4ddd-959f-388d81e646af.png)

**Comment by markerikson** [comment]
I'm sorry, I'm confused - I don't understand the actual question, and I don't understand what the reducer code is trying to accomplish in the first place.  Could you give more details?

**Reply by EdgarsLv** [reply]
> I'm sorry, I'm confused - I don't understand the actual question, and I don't understand what the reducer code is trying to accomplish in the first place. Could you give more details?

conversations = array with objects. currently length is 14
i'm taking conversations from state and trying to filter them by id. action.payload = id i pass do dispatch.

if i do:  `conversations.map((x) => x.id === action.payload)` 
then can see that it console logs correct result: array with 14 times boolean value where one value is true as it match x.id === action.payload.

With:  `conversations.filter((x) => x.id === action.payload)`
i should get console log with array with 1 object. But i'm not, i get proxy bla bla in console log.

in js i get correct value array with 1 object. In ts i get proxy bla bla.

**Reply by phryneas** [reply]
This will behave 100% exactly the same, no matter if you use JavaScript or TypeScript - when migrating one to the other, you don't change any runtime code, you only add type annotation.

That said, it is perfectly normal to get `Proxy` object if you `console.log(state)` in a slice reducer. That is covered in [debugging and inspecting state](https://redux-toolkit.js.org/usage/immer-reducers#debugging-and-inspecting-drafted-state).

Unfortunately, browsers are pretty bad in displaying `Proxy` objects. In RTK, to log your `state`, you would `console.log(current(state))` with `import { current } from '@reduxjs/toolkit'`.

Keep in mind though that you only need to do that for logging, not for manipulating!
